server.go

```diff
@@ -355,15 +367,33 @@ func (s *Server) StartAsSubserver(lndClient lnrpc.LightningClient,
 		}
 	}()
 
-	// Start the macaroon service and let it create its default macaroon in
-	// case it doesn't exist yet.
-	if err := s.startMacaroonService(createDefaultMacaroonFile); err != nil {
+	// Create and start the macaroon service and let it create its default
+	// macaroon in case it doesn't exist yet.
+	var err error
+	if s.macaroonService, err = lndclient.NewMacaroonService(
```

Comment:
nit: that `if` is easily overlooked IMO. Separate into assignment and then condition?